### ISSUE TITLE
Issue 24788: make mousewheel actions customizable

### ISSUE THREAD

**Issue by binarycoder** [TakingInitiative/wesnoth#24]
https://gna.org/bugs/?24788

On most mouses which which don't support horizonal scroll via mousehweel, the mousewheel scrolling is quite useless, since you still need other ways for the horizontal scolling. 
It'd be nice if it was possible to assign some useful action to it like: 
Increase/decrease accelerated speed
Cycle next/previous unit

**Comment by binarycoder** [comment]
Keybinding found in hotkey/hotkey_preferences_display
Scrolling found in mouse_events 480.

Should need to add exception or convert the existing scroll functions to hotkey functions, then allow the scroll up/down to be bound to hotkeys. Then update default keybindings to assign as before.

**Comment by binarycoder** [comment]
There are existing scroll hotkeys, but we will need to modify the hard coded ones in mouse_events to ignore it if they are used.

**Comment by michael-r-kolar** [comment]
We made a change that allows us to zoom in and out of the map, but as of right now it only has 2 zoom states. At first the program slides the screen up and down (vertically) by using the mouse wheel and one could slide the screen horizontally by holding either ALT key and moving the mouse wheel. So we made a change where only left ALT moves it horizontally, and now the right ALT allows the player to zoom in and out of the map. It's really odd that they used two buttons to do the same thing and that the default wheel event is wasted on something so useless, but those are design errors not technical bugs. 

change was made in:
 wesnoth/src/mouse_handler_base.cpp
line 273-274

this is the new code:
...
if (pressed[SDLK_LALT]) {
gui.scroll(-movey, -movex);
} else if (pressed[SDLK_RALT]) {
gui().set_default_zoom();
}else {
...

so we should change set_default_zoom() to something that allows more than the two zoom states of zoomed in and zoomed out. Also, it was noted in the comments that any changes to the zoom should leave the int divisible by 4, so the gui image isn't distorted.
